test/util/polkadot-api.ts

```diff
@@ -1,48 +1,96 @@
-import { createClient, type PolkadotClient } from "polkadot-api";
+import { createClient, type PolkadotClient, type TypedApi } from "polkadot-api";
 import { WebSocketProvider } from "polkadot-api/ws-provider/node";
 import { relaychain, storagehub } from "@polkadot-api/descriptors";
 
-export type TypesBundle = typeof relaychain | typeof storagehub;
+export type TypesBundle = typeof storagehub | typeof relaychain;
 
-export const getClient = async (endpoint: string, typesBundle: TypesBundle) => {
-  const client = createClient(WebSocketProvider(endpoint));
-  const api = client.getTypedApi(typesBundle);
-  const rt = await api.runtime.latest();
-  return { api, rt, client };
+type StorageHubApi = TypedApi<typeof storagehub>;
+
+export const waitForRandomness = async (api: StorageHubApi, timeoutMs = 60_000) => {
+  process.stdout.write("Waiting for randomness...");
+
+  const waitForValueOrTimeout = (timeoutMs: number): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      const timeout = setTimeout(() => {
+        reject(new Error("Timeout"));
+      }, timeoutMs);
+
+      let valueCount = 0;
+      const subscription = api.query.Randomness.LatestOneEpochAgoRandomness.watchValue(
+        "best"
+      ).subscribe((value) => {
+        valueCount++;
+
+        if (!value) {
+          reject(new Error("Randomness value is undefined"));
+          return;
+        }
+
+        if (valueCount === 2) {
```

Comment:
Why do you need to wait for `valueCount === 2`? Is it because you're waiting 2 epochs for randomness? In our parachain we're just waiting for one epoch.

Reply:
when you open a StorageValue subscription it returns an async `Observable` hook; to wait for the next epoch you need to have received two instances of data. This is because establishing a connection will immediately retrieve the current value and watch for updates.